zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/api/PurgeRequestTransformer.java

```diff
@@ -0,0 +1,104 @@
+import io.camunda.zeebe.dynamic.config.changes.ConfigurationChangeCoordinator.ConfigurationChangeRequest;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfiguration;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfigurationChangeOperation;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfigurationChangeOperation.PartitionChangeOperation.DeleteHistoryOperation;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfigurationChangeOperation.PartitionChangeOperation.PartitionBootstrapOperation;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfigurationChangeOperation.PartitionChangeOperation.PartitionJoinOperation;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfigurationChangeOperation.PartitionChangeOperation.PartitionLeaveOperation;
+import io.camunda.zeebe.dynamic.config.state.MemberState;
+import io.camunda.zeebe.util.Either;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+public final class PurgeRequestTransformer implements ConfigurationChangeRequest {
+
+  @Override
+  public Either<Exception, List<ClusterConfigurationChangeOperation>> operations(
+      final ClusterConfiguration clusterConfiguration) {
+
+    final SortedMap<Integer, PartitionBootstrapOperation> primaries =
+        createBootstrapOperations(clusterConfiguration.members());
+
+    final Map<Integer, List<PartitionJoinOperation>> followers =
+        new TreeMap<>(Comparator.naturalOrder());
+
+    final List<ClusterConfigurationChangeOperation> operations = new ArrayList<>();
+    for (final var member : clusterConfiguration.members().entrySet()) {
+      final var memberId = member.getKey();
+      for (final var partitions : member.getValue().partitions().entrySet()) {
+        final var partitionId = partitions.getKey();
+        operations.add(new PartitionLeaveOperation(memberId, partitionId));
+
+        final var primaryForPartition = primaries.get(partitionId);
+
+        if (!primaryForPartition.memberId().equals(memberId)) {
+          followers
+              .computeIfAbsent(partitionId, key -> new ArrayList<>())
+              .add(
+                  new PartitionJoinOperation(
+                      memberId, partitionId, partitions.getValue().priority()));
+        }
+      }
+    }
+
+    clusterConfiguration
+        .members()
+        .forEach((key, value) -> operations.add(new DeleteHistoryOperation(key)));
+
+    primaries.forEach(
+        (partitionId, bootstrapOperation) -> {
+          operations.add(bootstrapOperation);
+        });
+
+    followers.forEach(
+        (partitionId, joinOperations) -> {
+          operations.addAll(joinOperations);
+        });
+
+    return Either.right(operations);
+  }
+
+  /** This method creates the BootstrapOperations for all leaders for each partition. */
+  private SortedMap<Integer, PartitionBootstrapOperation> createBootstrapOperations(
```

Comment:
Yes, we select the primaries beforehand, because we don't know whether the current member is a primary if we have not seen the other partition members. Of course we could add it as primary, but when we encounter the next member and notice that this is actually the primary, we'd have to delete it from the primaries map and put it in the followers map and add the current as primary. That seemed more complex than figuring out the primaries beforehand.